Website/Providers/DataProviders/SqlDataProvider/07.04.02.SqlDataProvider

```diff
@@ -521,6 +521,174 @@ AS
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCacheAddItem
+	@ItemId int,
+	@CacheKey [varchar](36),
+	@Data	ntext,
+	@Expiration	datetime
+AS
+    INSERT INTO {databaseOwner}{objectQualifier}OutputCache
+		(ItemId, 
+		CacheKey, 
+		Data, 
+		Expiration)
+    VALUES
+		(@ItemId, 
+		@CacheKey, 
+		@Data, 
+		@Expiration)
+GO
+
+/* Add New OutputCacheGetItemCount  */
+/************************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCacheGetItemCount]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetItemCount
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetItemCount
+	@ItemId int
+AS
+    SELECT COUNT(*)
+    FROM {databaseOwner}{objectQualifier}OutputCache
+    WHERE ItemId = @ItemId
+GO
+
+/* Add New OutputCachePurgeCache  */
+/**********************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCachePurgeCache]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCachePurgeCache
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCachePurgeCache
+AS
+    DELETE
+    FROM {databaseOwner}{objectQualifier}OutputCache
+GO
+
+/* Add New OutputCachePurgeExpiredItems  */
+/*****************************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCachePurgeExpiredItems]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCachePurgeExpiredItems
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCachePurgeExpiredItems
+	@CurrentUtcDateTime datetime
+AS
+    DELETE
+    FROM {databaseOwner}{objectQualifier}OutputCache
+    WHERE Expiration <= @CurrentUtcDateTime
+GO
+
+/* Add New OutputCacheRemoveItem  */
+/**********************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCacheRemoveItem]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCacheRemoveItem
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCacheRemoveItem
+	@ItemId int
+AS
+    DELETE
+    FROM {databaseOwner}{objectQualifier}OutputCache
+    WHERE ItemId = @ItemId
+GO
+
+/* Add New OutputCacheGetKeys  */
+/*******************************/
+
+IF EXISTS (SELECT * FROM dbo.sysobjects WHERE id = object_id(N'{databaseOwner}[{objectQualifier}OutputCacheGetKeys]') AND OBJECTPROPERTY(id, N'IsPROCEDURE') = 1)
+  DROP PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetKeys
+GO
+
+CREATE PROCEDURE {databaseOwner}{objectQualifier}OutputCacheGetKeys
+	@ItemId int
+AS
+    DELETE
+    FROM {databaseOwner}{objectQualifier}OutputCache
+    WHERE ItemId = @ItemId or @ItemId IS NULL
```

Comment:
Do you really want to allow deletion of ALL keys here (disregarding portal etc?
Would it been helpful to use "WHERE ItemId Like @ItemID", this would support wildcards (at the end of the key) to delete a key set.